src/codeu/chat/client/ClientUser.java

```diff
@@ -88,16 +89,20 @@ public void showCurrent() {
     printUser(current);
   }
 
-  public void addUser(String name) {
-    final boolean validInputs = isValidName(name);
+  public void addUser(String name, String password) {
+    
+    final boolean validInputsName = isValidName(name);
+    final boolean validInputsPass = isValidName(password);
+    //TODO: Write isValidPassword(String password); method
+    //validInputs = isValidPassword(password);
 
-    final User user = (validInputs) ? controller.newUser(name) : null;
+    final User user = (validInputsName && validInputsPass) ? controller.newUser(name, password) : null;
```

Comment:
You should check out Optional. `final Optional<User> user = validInputsName && validInputsPass ? Optional.of(controller.newUser(name, password)) : Optional.absent()`